finetuner-workflow/finetuner/finetuner.py

```diff
@@ -362,79 +750,116 @@ def __getitem__(self, idx) -> Tuple[torch.Tensor, torch.Tensor]:
 else:
     bs = args.bs
 
+if is_main_process():
+    logger.info(f"BS: {bs}")
+    logger.info(f"GAS: {args.gradients}")
+    devices_repr = device
+    if device != "cpu":
+        try:
+            devices_repr = (f"{torch.cuda.device_count()}x"
+                            f"{torch.cuda.get_device_name(0)}")
+        except Exception:
+            pass
+    logger.info(f"DEVICES: {devices_repr}")
+
 # Rewrite our `ds_config` to match arguments passed in.
-ds_args  = {}
+ds_args = {}
 if device != "cpu":
     ds_config = json.load(open(args.ds_config))
-    if "zero_optimization" in ds_config and \
-            ds_config["zero_optimization"].get("stage", None) != \
-            args.zero_stage:
+    if "zero_optimization" in ds_config:
         ds_config["zero_optimization"]["stage"] = args.zero_stage
-    ds_args['deepspeed'] = ds_config
+    ds_args["deepspeed"] = ds_config
+    ds_args["local_rank"] = args.local_rank
+    os.environ["LOCAL_RANK"] = str(args.local_rank)
 else:
-    ds_args['no_cuda'] = True
-    ds_args['local_rank'] = -1
+    ds_args["no_cuda"] = True
+    ds_args["local_rank"] = -1
     os.environ["LOCAL_RANK"] = "-1"
 
+# Print our deepspeed estimates
+estimate_fn = None
+if args.zero_stage in [1, 2] and is_main_process():
+    logger.info("DeepSpeed ZeRO-1/2 Memory Estimates")
+    estimate_fn = estimate_zero2_model_states_mem_needs_all_live
+elif args.zero_stage == 3 and is_main_process():
+    logger.info("DeepSpeed ZeRO-3 Memory Estimates")
+    estimate_fn = estimate_zero3_model_states_mem_needs_all_live
+if estimate_fn:
+    estimate_fn(model=model,
+                num_nodes=1,
+                num_gpus_per_node=torch.cuda.device_count())
+
+# The latest deepspeed logging is pretty obnoxious, so we disable it.
+deepspeed.utils.logger.setLevel(logging.WARNING)
```

Comment:
The `deepspeed` logger is a bit obnoxious but it does also provide a lot of useful information during initialization, enough that I found myself re-enabling it for most of the work I was doing on this script. It might be good to control this behaviour through a command line flag like the regular logging level, or at least to re-enable this whenever `args.log_level.upper() == "DEBUG"`.